src/main/java/com/redhat/devtools/lsp4ij/usages/LSPUsageSearcher.java

```diff
@@ -98,7 +115,12 @@ public void processElementUsages(@NotNull PsiElement element, @NotNull Processor
         }
 
         // For completeness' sake, also collect external usages to LSP (pseudo-)elements
-        LSPExternalReferencesFinder.processExternalReferences(file, element.getTextOffset(), reference -> processor.process(new UsageInfo2UsageAdapter(new UsageInfo(reference))));
+        LSPExternalReferencesFinder.processExternalReferences(
+                file,
+                ReadAction.compute(element::getTextOffset),

```

Comment:
This is the specific call that was causing the exception in Joachim's report. Now we get the text offset in a read action.